internal/net/conn_other.go

```diff
@@ -0,0 +1,58 @@
+//go:build test
+
+package net
+
+import (
+	"context"
+
+	"github.com/drand/drand/v2/internal/metrics"
+	"go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/connectivity"
+	insecure "google.golang.org/grpc/credentials/insecure"
+)
+
+// conn retrieve an already existing conn to the given peer or create a new one
+func (g *grpcClient) conn(ctx context.Context, p Peer) (*grpc.ClientConn, error) {
+	g.Lock()
+	defer g.Unlock()
+	var err error
+
+	// we try to retrieve an existing connection if available
+	c, ok := g.conns[p.Address()]
+	if ok && c.GetState() == connectivity.Shutdown {
```

Comment:
Yeah, the problem with `TransientFailure` was that it was basically a trap for connections that had fallen back to non-TLS connection: they got stuck there because then `ok` was always true but with a bad connection that was trying to talk non-TLS to the server which expected TLS.

Now in theory we can count on the auto-reconnect to do what it's supposed to do.